ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -39,9 +39,214 @@
+                addFormParameter(afAssetPath + '/schemaRef');
+            }
+            if (schemaType != NONE) {
+                document.querySelector(FORM_MODEL_SELECTOR).disabled = true;
+                dialog.find(SCHEMA_TYPE)[0].removeAttribute("disabled");
+            }
+            document.body.appendChild(formModelChangeConfirmationDialog);
+            prefillSchema(schemaType, dialog);
+        }
+    };
+
+    function selectFormModelOnChanged(dialog) {
+        var schemaTypeSelected = dialog.find(FORM_MODEL_SELECTOR);
+        if(schemaTypeSelected.length > 0){
+            schemaTypeSelected = schemaTypeSelected[0].value;
+            setElementValue(dialog, DAM_SCHEMA_TYPE, schemaTypeSelected)
+            hideContainersExcept(schemaTypeSelected);
+        }
+    };
+
+    function prefillSchema(schemaType, dialog){
+        var schemaRef = dialog.find(SCHEMA_REF);
+        if(schemaRef.length > 0){
+            schemaRef = schemaRef[0].value;
+            setElementValue(dialog, DAM_SCHEMA_REF, schemaRef);
+            if (schemaType == JSON_SCHEMA) {
+                $(SCHEMA_DROPDOWN_SELECTOR).val(schemaRef);
+            } else if (schemaType == FORM_DATA_MODEL) {
+                $(FDM_DROPDOWN_SELECTOR).val(schemaRef);
+            }
+        }
+    };
+
+    function schemaSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(SCHEMA_DROPDOWN_SELECTOR);
+        if(selectedSchema.length > 0){
+            selectedSchema = selectedSchema[0].value;
+            setElementValue(dialog, SCHEMA_REF, selectedSchema);
+            setElementValue(dialog, DAM_SCHEMA_REF, selectedSchema);
+            if (configuredFormModel) {
+                confirmFormModelChange(selectedSchema, $(SCHEMA_DROPDOWN_SELECTOR));
+            } else {
+                toBeConfiguredFormModel = selectedSchema;
+            }
+        }
+    };
+
+    function fdmSelectorOnChanged(dialog) {
+        var selectedSchema = dialog.find(FDM_DROPDOWN_SELECTOR);
+        if(selectedSchema.length > 0) {
+            selectedSchema = selectedSchema[0].value;
+            setElementValue(dialog, SCHEMA_REF, selectedSchema);
+            setElementValue(dialog, DAM_SCHEMA_REF, selectedSchema);
+            if (configuredFormModel) {
+                confirmFormModelChange(selectedSchema, $(FDM_DROPDOWN_SELECTOR));
+            } else {
+                toBeConfiguredFormModel = selectedSchema;
+            }
+        }
+    };
+
+    function setElementValue(dialog, elementRef, value){
+        var element = dialog.find(elementRef);
+        if(element.length > 0){
+            element[0].value = value;
+        }
+    }
+
+    function getAfAssetMetadataPath() {
+        return window.guidelib.author.AuthorUtils.getGuideContainerPath().replace(FM_AF_ROOT, FM_DAM_ROOT).replace("/guideContainer", "/metadata");
+    }
+
+    function isForm(){
+        return Granite.author.page.path.startsWith(FM_AF_ROOT);
```

Comment:
we should check for the presence of /content/dam/.. path corresponding to the form, and then take decision.
pls avoid checking for forms vs sites editor in code.

Reply:
@nit23uec, if we are checking it for every page than we are not sure what to replace in the path.
let's understand it through an example: 
AF path for a container will be like: /content/forms/af/v2-t1/jcr:content/guideContainer
it's corresponding dam path will be: /content/dam/formsanddocuments/v2-t1/jcr:content/guideContainer

but for sites editor or any other form it's path can be anything like: 
/content/wknd/us/en/about-us/jcr:content/root/container/container_1358206273 

so, where should I check?
/content/dam/formsanddocuments/en/about-us/jcr:content/root/container/container_1358206273 
or
/content/dam/formsanddocuments/wknd/en/about-us/jcr:content/root/container/container_1358206273 
or
/content/dam/formsanddocuments/wknd/us/en/about-us/jcr:content/root/container/container_1358206273 


This can be achieved via exposing a GET API and checking through select query at backend but I guess that would be over engineering for it.
cc: @ujjwal5

Reply:
ack. I was thinking abt: /`content/dam/formsanddocuments/wknd/us/en/about-us/jcr:content/root/container/container_1358206273` but it's fine for now if no other way seems good.
lets go with the currrent.